src/combinatorics/simple_graph/regularity/bound.lean

```diff
@@ -141,9 +138,26 @@ noncomputable def bound : ℕ :=
 (step_bound^[⌊4 / ε^5⌋₊] $ initial_bound ε l) * 16 ^ (step_bound^[⌊4 / ε^5⌋₊] $ initial_bound ε l)
 
 lemma initial_bound_le_bound : initial_bound ε l ≤ bound ε l :=
-(id_le_iterate_of_id_le le_step_bound _ _).trans $ nat.le_mul_of_pos_right $ pow_pos (by norm_num) _
+(id_le_iterate_of_id_le le_step_bound _ _).trans $ nat.le_mul_of_pos_right $ by positivity
 
 lemma le_bound : l ≤ bound ε l := (le_initial_bound ε l).trans $ initial_bound_le_bound ε l
 lemma bound_pos : 0 < bound ε l := (initial_bound_pos ε l).trans_le $ initial_bound_le_bound ε l
 
 end szemeredi_regularity
+
+namespace tactic
+open positivity szemeredi_regularity
+
+/-- Extension for the `positivity` tactic: `szemeredi_regularity.initial_bound` and
+`szemeredi_regularity.bound` are always positive. -/
+@[positivity]
```

Comment:
I thought we agreed this should be local attribute `positivity` rather than global? I agree this is a useful extension for things specific to regularity, but as soon as this file is imported anywhere, I don't think it's sensible for `positivity` to be even attempting matches

Reply:
No, the local one is [here](https://github.com/leanprover-community/mathlib/compare/szemeredi?expand=1#diff-c984e39a1b84c09c521d7d6d1aaadacc9d6f5538544aaa240469324a037e2dc1R60-R72) and proves `0 < n / step_bound P.parts.card` and `100 ≤ 4 ^ P.card * ε ^ 5 → 0 < ε`.

Reply:
Many positivity extensions apply rarely, like `nat.asc_factorial` or `ennreal.to_real`. Following this precedent, I think it's fine having this one be global for the same reason, especially given that we are only expecting a few files to import the present.

Reply:
Matching on identifiers is pretty cheap so I don't think we need to worry much about making this a local attribute. (Of course if you have profiler output saying otherwise, I'd be happy to change my mind!)